pkg/actions/lua/databricks/client.go

```diff
@@ -0,0 +1,183 @@
+	if err := validateTableInput(tableName, location); err != nil {
+		return "", errors.Join(ErrTableCreationFailure, err)
+	}
+	statement := fmt.Sprintf(`CREATE EXTERNAL TABLE %s LOCATION '%s'`, tableName, location)
+	esr, err := dbc.workspaceClient.StatementExecution.ExecuteAndWait(dbc.ctx, sql.ExecuteStatementRequest{
+		WarehouseId: warehouseID,
+		Catalog:     catalogName,
+		Schema:      schemaName,
+		Statement:   statement,
+	})
+	if err != nil {
+		return "", errors.Join(ErrTableCreationFailure, err)
+	}
+	return esr.Status.State.String(), nil
+}
+
+func tableFullName(catalogName, schemaName, tableName string) string {
+	return fmt.Sprintf("%s.%s.%s", catalogName, schemaName, tableName)
+}
+
+func (dbc *Client) dropTable(catalogName, schemaName, tableName string) error {
+	err := dbc.workspaceClient.Tables.DeleteByFullName(dbc.ctx, tableFullName(catalogName, schemaName, tableName))
+	if err != nil {
+		return errors.Join(ErrTableDeletionFailure, err)
+	}
+	return nil
+}
+
+func (dbc *Client) createOrGetSchema(catalogName, schemaName string) (*catalog.SchemaInfo, error) {
+	schemaInfo, err := dbc.workspaceClient.Schemas.Create(dbc.ctx, catalog.CreateSchema{
+		Name:        schemaName,
+		CatalogName: catalogName,
+	})
+	if err != nil {
+		if strings.Contains(err.Error(), "already exists") {
+			// Full name of schema, in form of <catalog_name>.<schema_name>
+			schemaInfo, err = dbc.workspaceClient.Schemas.GetByFullName(dbc.ctx, catalogName+"."+schemaName)
+			if err != nil {
+				return nil, err
+			}
+			return schemaInfo, nil
+		}
+		return nil, err
+	}
+	return schemaInfo, nil
+}
+
+func newDatabricksClient(l *lua.State) (*databricks.WorkspaceClient, error) {
+	host := lua.CheckString(l, -2)
+	token := lua.CheckString(l, -1)
+	wsc, err := databricks.NewWorkspaceClient(
+		&databricks.Config{
+			Host:        host,
+			Token:       token,
+			Credentials: config.PatCredentials{},
+		},
+	)
+	if err != nil {
+		return nil, err
+	}
+	return wsc, err
+}
+
+func registerExternalTable(client *Client) lua.Function {
+	return func(l *lua.State) int {
+		tableName := lua.CheckString(l, 1)
+		location := lua.CheckString(l, 2)
+		warehouseID := lua.CheckString(l, 3)
+		catalogName := lua.CheckString(l, 4)
+		schemaName := lua.CheckString(l, 5)
+
+		status, err := client.createExternalTable(warehouseID, catalogName, schemaName, tableName, location)
+		if err != nil {
+			if strings.Contains(err.Error(), "already exists") {
+				err = client.dropTable(catalogName, schemaName, tableName)
+				if err != nil {
+					lua.Errorf(l, err.Error())
+					panic("unreachable")
+				}
+				status, err = client.createExternalTable(warehouseID, catalogName, schemaName, tableName, location)
+				if err != nil {
+					lua.Errorf(l, err.Error())
+					panic("unreachable")
+				}
+			} else {
+				lua.Errorf(l, err.Error())
+				panic("unreachable")
+			}
+		}
+		l.PushString(status)
+		return 1
+	}
+}
```

Comment:
Still think this is an issue that we enable this type of operation - can we discuss other options.
+ Document this option if there is no other way

Reply:
sure we can discuss